src/DefaultRESTClient.ts

```diff
@@ -206,10 +206,18 @@ export default class DefaultRESTClient<RT, ERT> implements IRESTClient<RT, ERT>
   }
 
   private getQueryString() {
-    var queryString = '';
-    for (let key in this.parameters) {
+    let queryString = '';
+    const appendParam = (key: string, param: string) => {
       queryString += (queryString.length === 0) ? '?' : '&';
-      queryString += key + '=' + encodeURIComponent(this.parameters[key]);
+      queryString += key + '=' + encodeURIComponent(param);
```

Comment:
The key may also contain values that need to be URL encoded. 

I think the correct pattern is: 

```
queryString += encodeURIComponent(key) + '=' + encodeURIComponent(param);
```